src/main/kotlin/com/mdrobnak/lalrpop/psi/util/LpPsiFoldUtil.kt

```diff
@@ -0,0 +1,27 @@
+package com.mdrobnak.lalrpop.psi.util
+
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.elementType
+import com.mdrobnak.lalrpop.psi.LpElementTypes
+import org.rust.lang.core.psi.ext.childrenWithLeaves
+import org.rust.lang.core.psi.ext.endOffset
+import org.rust.lang.core.psi.ext.startOffset
+
+fun depthOneBraceMatcherFoldDescriptors(element: PsiElement, document: Document, quick: Boolean): List<FoldingDescriptor> {
```

Comment:
The extern and enum elements which use this function don't show up as collapsible in the IDE. It might be an issue with the algorithm. It looks like it might generate invalid descriptors. Example input:
```
extern {
    // ...
    enum lexer::Tok {
        // ...
    }
}
```
Output descriptors are from the second LBrace to the first RBrace (good) and from the second LBrace to the second RBrace (bad).

Match elements do show up as collapsible, but they only have one set of braces so the algorithm seems to work in that case.

Maybe you could also add some documentation to clarify what this function is supposed to output (i.e. why it has "depthOne" in the name).